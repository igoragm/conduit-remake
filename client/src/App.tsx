import React from "react";
import "./App.scss";
import Header from "./Components/Modules/Header/Header";
import { Switch, Route } from "react-router";
import Home from "./Pages/Home/Home";
import Article from "Pages/Article/Article";

function App() {
    return (
        <div id="app">
            <Header />
            <Switch>
                <Route exact={true} path="/" component={Home} />
                <Route exact={true} path="/article/:id" component={Article} />
            </Switch>
        </div>
    );
}

export default App;
