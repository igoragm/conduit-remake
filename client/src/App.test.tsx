import React from "react";
import App from "./App";
import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";

describe("App Test Suite", () => {
    let wrapper: ShallowWrapper<typeof App>;

    beforeEach(() => {
        wrapper = shallow(<App />);
    });

    test("App snapshot renders", () => {
        const appJson = toJson(wrapper);

        expect(appJson).toMatchSnapshot();
    });

    test("App component ID", () => {
        const appComponent = wrapper.find("#app");

        expect(appComponent).toHaveLength(1);
    });

    afterEach(() => {
        wrapper.unmount();
    });
});
