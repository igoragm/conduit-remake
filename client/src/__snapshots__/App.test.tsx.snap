// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Suite App snapshot renders 1`] = `
<div
  id="app"
>
  <Header />
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Home)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/article/:id"
    />
  </Switch>
</div>
`;
